integration_tests/src/main/python/hash_aggregate_test.py

```diff
@@ -2539,3 +2540,27 @@ def run_and_capture_plan(spark, get_df_function):
     # two-stage aggregate should be 2x the number of logical aggregates
     assert aggregate_nodes == 2, "Unexpected SparkPlan: " + plan_str
     assert_gpu_and_cpu_are_equal_collect(run_spark_fn, conf=run_conf)
+
+
+@pytest.mark.skipif(is_before_spark_350(),
+                    reason='bitwise operations are available from Spark 3.5.0')
```

Comment:
How does the code compile for 321/etc if it is only available after 3.5.0? Were they not added to SQL or what?

Spark code says it has been around since 3.0.0 for all of the aggregations added here?

https://github.com/apache/spark/blob/6ac21cf9ad3a47c26e8498b3a915800f93b0fdc3/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/bitwiseAggregates.scala#L64